sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -457,7 +457,6 @@ case class NewInstance(
     }
     outerPointer.map { p =>
       val outerObj = p()
-      val d = outerObj.getClass +: paramTypes
       val c = getConstructor(outerObj.getClass +: paramTypes)
```

Comment:
here, we use `outerObj.getClass +: paramTypes` directly instead of the `d` at line 460.

Reply:
Ah, ok. I've checked this var is not used.